addon/lib/dependencies.js

```diff
@@ -0,0 +1,134 @@
+import { assert } from "@ember/debug";
+import { computed, get } from "@ember/object";
+
+import { getAST, getTransforms } from "ember-caluma/utils/jexl";
+
+/**
+ * Extract dependency slugs from a jexl expression.
+ *
+ * E.g: 'foo'|answer in 'bar'|answer|mapby('column') would result in the
+ * following dependency slugs: ['foo', 'bar', 'bar.column']
+ *
+ * @param {Jexl} jexl
+ * @param {String} expression
+ * @return {String[]} Returns an array of dependency slugs
+ */
+export function getDependenciesFromJexl(jexl, expression) {
+  const allTransforms = getTransforms(getAST(jexl, expression));
+  const answerTransforms = allTransforms.filter(
+    (transform) => transform.name === "answer"
+  );
+  const mapbyTransforms = allTransforms.filter(
+    (transform) =>
+      transform.name === "mapby" &&
+      transform.args[0].type === "FunctionCall" &&
+      transform.args[0].name === "answer"
+  );
+
+  return [
+    ...new Set([
+      ...answerTransforms.map((transform) => transform.args[0].value),
+      ...mapbyTransforms.map(
+        (transform) =>
+          `${transform.args[0].args[0].value}.${transform.args[1].value}`
+      ),
+    ]),
+  ];
+}
+
+/**
+ * Find a certain field in a document or throw an informative error.
+ *
+ * @param {Document} document The document containing the field
+ * @param {String} slug The question slug of the searched field
+ * @param {String} expression The expression to append to the error message
+ * @return {Field} The searched field
+ */
+export function findField(document, slug, expression) {
+  const field = document.findField(slug);
+
+  assert(
+    `Field for question \`${slug}\` could not be found in the document \`${document.uuid}\`. Please verify that the jexl expression is correct: \`${expression}\`.`,
+    field
+  );
+
+  return field;
```

Comment:
Are there cases where we're calling `document.findField` with non-existing slugs but don't want the error? If not, I'd move the assertion into `document.findField` and drop this method.

Reply:
That would be a breaking change since `findField` is public API that is used in various projects. They depend on it returning undefined if the field is not found. Asserting something there is not a good idea IMO

Reply:
:+1: Fair enough!